bin/broadcast_prs.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+from cookiecutter.main import cookiecutter
+import git
+import json
+import os
+import requests
+from requests.auth import HTTPBasicAuth
+import shutil
+import sys
+import subprocess
+import tempfile
+import utils
+
+# The GitHub base url or the nf-core project
+GH_BASE_URL = "https://{token}@github.com/nf-core"
+# The current cookiecutter template url for nf-core pipelines
+NF_CORE_TEMPLATE = os.path.join(
+                        os.path.dirname(
+                                os.path.dirname(os.path.realpath(__file__))
+                        ), "nf_core/pipeline-template")
+# The JSON file is updated on every push event on the nf-core GitHub project
+NF_CORE_PIPELINE_INFO = "http://nf-co.re/pipelines.json"
+# The API endpoint for creating pull requests
+GITHUB_PR_URL_TEMPL = "https://api.github.com/repos/nf-core/{pipeline}/pulls"
+
+
+class UpdateTemplate:
+    """Updates the template content of an nf-core pipeline in
+    its `TEMPLATE` branch.
+
+    Args: - pipeline: The pipeline name
+          - branch: The template branch name, default=`TEMPLATE`
+          - token: GitHub auth token
+    """
+
+    def __init__(self, pipeline, branch='TEMPLATE', token=""):
+        """Basic constructor
+        """
+        self.pipeline = pipeline
+        self.repo_url = "{base_url}/{pipeline}".format(
+                    base_url=GH_BASE_URL.format(token=token),
+                    pipeline=pipeline)
+        self.branch = branch
+        self.tmpdir = tempfile.mkdtemp()
+        self.templatedir = tempfile.mkdtemp()
+        self.repo = None
+    
+    def run(self):
+        """Execute the template update.
+        """
+        config = self._clone_repo()
+        assert self.repo
+        context = utils.create_context(config)
+        self._apply_changes(context)
+        self._commit_changes()
+        self._push_changes()
+
+    def _clone_repo(self):
+        """Clone the repo and switch to the configured branch.
+        """
+        self.repo = git.Repo.clone_from(self.repo_url, self.tmpdir)
+        config = utils.fetch_wf_config(wf_path=self.tmpdir)
+        self.repo.git.checkout("origin/{branch}".format(branch=self.branch),
```

Comment:
I guess this will fail when pipelines doesn't have a TEMPLATE branch. Might be useful to fail gracefully in that case, since it will be very common at least in the beginning?

Reply:
I think failing here is totally ok, as the error will be listed in the Travis log anyway with stacktrace. So in my opinion no need to try catch here.

Reply:
Yes sure, but it would also cause the script to crash on the first pipeline without a TEMPLATE branch and no pull requests at all would be created. But that is perhaps what you intended?

Reply:
I mean, i can catch the errors in a list, and print them afterwards? So we see all failing pipelines at once?

Reply:
Ah think this indeed would make sense here

Reply:
Sorry, I didn't understand if you changed your mind or not. From my understanding the script will be run once in the tools travis and it would be strange if it would fail/crash until *all* pipelines are set up properly, right?

Reply:
I think he changed his mind to agree with you 😉

Reply:
yup, I agree with @alneberg 's suggestion